---
title: "Music AD RNASeq"
author: "Christine Lucille Kuryla (clk2162)"
date: "2023-11-29"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(DESeq2)
library(knitr)
```

## Load and explore the data

First, load the data. 

```{r load_data, message = FALSE}
# metadata
metadata_raw <- read.table("./data_edited/music_ad_copyandpaste.txt",
                           col.names = 
                             c("gse_number",
                             "group",
                             "timepoint",
                             "subj_id",
                             "sample_id")
                           ) 
metadata <- metadata_raw %>% 
  mutate(group = str_sub(group, 1, -2)) %>% 
  mutate(timepoint = str_sub(timepoint, 1, -2)) %>% 
  mutate(group = as.factor(group),
         timepoint = as.factor(timepoint)) %>% 
  mutate(group = fct_relevel(group, "Control", "ACD")) %>% 
  mutate(timepoint = fct_relevel(timepoint, "Timepoint-1", "Timepoint-2"))

# raw count data
countdata_raw <- read_tsv("./internet_files/GSE239282_raw_counts_GRCh38.p13_NCBI.tsv") 

countdata <- countdata_raw %>% 
  column_to_rownames("GeneID")
```


```{r filtering_data, message = FALSE}
# Not all of the samples in the metadata file are in the RNASeq countdata file, so we will filter the metadata

# Filter to only have consistent subjects. The subjects are not consistent in the pre and post music groups (timepoint 1 and timepoint 2).

acd_1 <- metadata %>% 
  filter(group == "ACD") %>% 
  filter(timepoint == "Timepoint-1") %>% 
  pull(subj_id)

acd_2 <- metadata %>% 
  filter(group == "ACD") %>% 
  filter(timepoint == "Timepoint-2") %>% 
  pull(subj_id)

control_1 <- metadata %>% 
  filter(group == "Control") %>% 
  filter(timepoint == "Timepoint-1") %>% 
  pull(subj_id)

control_2 <- metadata %>% 
  filter(group == "Control") %>% 
  filter(timepoint == "Timepoint-2") %>% 
  pull(subj_id)

acd_intersect <- intersect(acd_1, acd_2)
control_intersect <- intersect(control_1, control_2)

final_subject_set <- c(acd_intersect, control_intersect)

# filter 

metadata_filtered <- metadata %>% 
  filter(gse_number %in% colnames(countdata)) %>% 
  filter(subj_id %in% final_subject_set)

countdata_filtered <- countdata %>% 
  select(all_of(pull(metadata_filtered, gse_number)))

metadata <- metadata_filtered
countdata <- countdata_filtered

```

```{r explore_data}

summary_acd <- metadata %>% 
  filter(group == "ACD") %>% 
  group_by(timepoint) %>% 
  summarize(n_acd = n())

summary_control <- metadata %>% 
  filter(group == "Control") %>% 
  group_by(timepoint) %>% 
  summarize(n_control = n())

inner_join(summary_acd, summary_control, by = "timepoint") %>% 
  knitr::kable()

countdata %>% 
  summarise(n_transcripts = n()) %>% 
  knitr::kable()

```

There seems to be something that is separating the groups more than condition or timepoint. It is likely a batch effect. I don't think the rest of this analysis is worthwhile; QC has failed.


## Former Analysis Plan

For our first analysis, we will determine the differential transcript expression between the ACD subjects and the controls at each time point. This will help us see if the two groups (ACD and control) have different gene expression (regardless of the intervention). 

Four our second analysis, we will determine the differential transcript expression between Timepoint 2 (after music) and Timepoint 1 (baseline - before music) for both the AD and control subjects. This will show us the difference in their responses to the intervention. 

*Analysis 1 -- ACD Subjects vs Controls (for each timepoint separately)*

We will begin by loading and exploring the data. First, we will normalize and/or perform a variance stabilized transformation on the counts in order to visualize and explore group separation using PCA, heatmaps, and dendrograms. Then, we will perform a differential expression (DE) analysis for ACD Subjects vs Controls (for each timepoint separately). We will explore which genes are significantly differentially expressed (up or down regulated), visualize with another heatmap and volcano plot, perform gene ontology (GO) analysis, pathway analysis, and GSEA.

*Analysis 2 -- After Music vs Before Music (for ACD and controls separately)*

We will perform an analogous analysis, except this time we will explore pre- and post- music separation an differential expression. 

*Combined Analysis -- Relating and Interpreting Results*

We will look at overlapping differentially expressed genes, compare the various analyses, and interpret the differences.

## Define Functions

We will perform multiple similar analyses, so will begin by defining some functions. 

```{r define_functions}

dds <- DESeqDataSetFromMatrix(countData = countdata,
                              colData = metadata,
                              design= ~ group)

  # Visualize the effect of the filtering
    # Before filtering
    hist(log1p(counts(dds)), main = "Before filtering", xlab = "log(counts + 1)", breaks = 50)
    
    # After filtering 
  #  keep <- rowSums(counts(dds)) >= 10
    keep <- rowSums(assay(dds) >= 5) >= 4
    dds_filtered <- dds[keep,]
    hist(log1p(counts(dds_filtered)), main = "After filtering", xlab = "log(counts + 1)", breaks = 50)

  
    dds <- dds_filtered
    
  # relevel (not necessary for the unsupervised analysis but good for future object DE analysis)
  # make sure reference is set to the reference level for interpretablity 
  relevel(dds$group, ref = "Control") 
  
  # Estimate size factors, accounts for library depth and composition (1 and 3)
  # This estimates the size factors and inserts them into the dds object so we can use to normalize
  dds <- estimateSizeFactors(dds)
  sizeFactors(dds)
    
dds <- DESeq(dds)
resultsNames(dds) # lists the coefficients
res <- results(dds, name="group_ACD_vs_Control")
# or to shrink log fold changes association with condition:
res <- lfcShrink(dds, coef="group_ACD_vs_Control", type="apeglm")


vsd <- vst(dds, blind=TRUE)
plotPCA(vsd, intgroup="group") +
  ggtitle("Music and AD colored by Condition")

plotPCA(vsd, intgroup="timepoint") +
  ggtitle("Music and AD colored by Timepoint")

plotPCA(vsd, intgroup="group", pcsToUse = 2:3) 
plotPCA(vsd, intgroup="timepoint", pcsToUse = 2:3) 

```









## Analysis 2 - comparing before and after music

